### ISSUE TITLE
Support for Ktor 2.0.0

### ISSUE THREAD

**Issue by Marek00Malik** [papsign/Ktor-OpenAPI-Generator#128]
Hey, 
When can we count on support for the new Ktor release -> 2.0.0 ?
I'm not able to install the OpenAPIGen plugin.

**Comment by vladtrc** [comment]
In my opinion such functionality should already be built in...

**Comment by Wicpar** [comment]
i don't work with Ktor anymore, i can't spare the time for such a rework, however i will accept pull requests...

**Comment by LukasForst** [comment]
Hey, I forked this repo and wired stuff together to work with Ktor 2.0.0 the fork is here - https://github.com/LukasForst/ktor-openapi-generator - it's also deployed on maven central, so feel free to use that build.

It's not very nice, Ktor removed some stuff this lib relied on (content type handling for example [see this ugly hack](https://github.com/LukasForst/ktor-openapi-generator/blob/3a7cf6653723656b321baae7c387502ea89090a5/src/main/kotlin/com/papsign/ktor/openapigen/content/type/ktor/KtorContentProvider.kt#L45) ) but I monkey patched most of the stuff and now I'm successfully using that in two of my projects in production with Ktor 2.0.0 - _so I guess the subset of stuff I use works_.

I'm not sure if we can upstream this to master here as during the process I migrated to gradle kotlin DSL and I'm not sure what plans does @Wicpar have with this project.

**Comment by Wicpar** [comment]
@LukasForst if you are willing to maintain the project i can add you as a collaborator, and it would be a good idea to update to the newer version, even if it's a bit hacky

**Comment by Marek00Malik** [comment]
Hey @LukasForst I'm trying out your fork actually now. The sad part is that there is no support for Location API which looks like the way Ktor is heading (I saw an issue even raised, but it got closed... #70). Are you @LukasForst planning to play with it? I know there is a repo called `kompendium` that is supporting it, but they are not on Ktor 2.0.0 yet.

**Comment by sigmanil** [comment]
@LukasForst Are you using StatusPages in your applications with your fork? I'm trying out your fork and I seem to have some trouble with statuspages (exception handling) in tests. It's a bit weird, the tests have gone unstable, the status pages set up using the OpenApiGen withApi syntax seem to trigger only sometimes.

**Comment by LukasForst** [comment]
hey @sigmanil can you try to update to latest Ktor 2.0.1? See https://youtrack.jetbrains.com/issue/KTOR-4187 because that was for example problem in my apps

actually, I'm surprised base ktor status pages work for you in 2.0.0 because according to that Ktor bug, it was broken (and it was broken in my services as well)

**Comment by LukasForst** [comment]
@Wicpar I'd like to, but definitely not now (busy on other projects, maybe during summer? I'll let you know). However, there're some rumours that JetBrains will work on Ktor Swagger as well (the issue has moved to "in progress") so maybe we will get native support. That would definitely be nice.

**Comment by LukasForst** [comment]
@Marek00Malik sorry, but right now, I don't have enough time to work on this, I just ported the basic stuff to keep my own service up and running and it is ugly hack anyway. Maybe I will get some time during summer, but unfortunately I can not guarantee anything.

**Comment by LukasForst** [comment]
looks like we're going to get official support after all - https://youtrack.jetbrains.com/issue/KTOR-774 - issue just changed from `Open` to `In Progress`

**Comment by darkxanter** [comment]
It looks like this library is still relevant as JetBrains makes OpenAPI spec generation a paid feature of IDEA Ultimate.

**Comment by LukasForst** [comment]
I agree, their solution is far from optimal. I guess we offload serving the UI to their implementation, but honestly even that one is worse, because it fetches js from remote source during runtime...

Anyway, there are now more alternatives in the wild such as https://github.com/bkbnio/kompendium and https://github.com/SMILEY4/ktor-swagger-ui 

My [fork](https://github.com/LukasForst/ktor-openapi-generator) of this repo is not exactly maintained, but I keep releasing dependency updates to maven central.

**Comment by Wicpar** [comment]
> My [fork](https://github.com/LukasForst/ktor-openapi-generator) of this repo is not exactly maintained, but I keep releasing dependency updates to maven central.

well this library isn't either, so if you want i can link to your repo in the readme for 2.0 compat.
